docs_src/design/adr/0019-Service-List.md

```diff
@@ -0,0 +1,87 @@
+- Generation of security tokens for the new services under most of the conceived alternatives is complex.  In the current security bootstrapping, secret store setup will require a restart of all the services in order to add or remove a service.  The ORRA project and some adopters are clear that they would like to have the ability to add/remove services dynamically at runtime.
+- From a security perspective, being able to add/remove new services dynamically opens up a rather large attack vector and vulnerability.
+
+### Implementation Alternative Discussion
+
+- A service list could be made available in a configuration file in the file system (non-Docker) or volume mount (in a Docker situation).
+  - All services would get the configuration in a consistent way (#1)
+  - It would also mean that there is a single, authoritative list for all services and it could be made to be ACID safe (#2 and #3).
+  - The file would have to be validated whenever edited in order not to bring down all the services with bad contents.
+  - This alternative would solve the requirement that the solution operate with or without Consul (#4).  UI and CLI should have access to the same file or volume mount.
+  - There is a question about who or what service would put this file into the file system or volume mount – especially since security bootstrapping would need it first but this file would be needed in non-secure runtime as well.
+  - There isn’t a central “dummy” service that would own the file – especially for dev/test.
+  - It has to be a service that is always there (like core metadata), but also available from the start of all services (which metadata is not in the case of security services).
+  - Services would have to monitor the file or volume for changes in order to be able to address the add/removal of services at runtime.
+  - How would the security bootstrapping address the add/removal of services and get new tokens established?  The bootstrapping service isn’t even long running to do this.
+- Have global configuration in Consul (or config/reg service) that everyone reads.
+  - When Consul is not used, have a common service list in each service configuration.toml file that everyone reads.
+  - While the means to get the configuration is consistent in this solution alternative (#1), there is not a single, authoritative list of services when using local configuration and making ACID transactions (#3) across the multiple files would be difficult – especially if distributed across hosts.
+  - Using Consul would make sure that the service list is valid and that any change operation is ACID.
```

Comment:
> service list is valid

What constitutes a validity when talking about a service list?  XML, for example, be both valid and well-formed.  Well-formed-ness means that it can be read by an XML parser.  Valid means that it confirms to a known schema.  You can make XML that is well-formed but not valid.

> any change operation is ACID

Depends on how a transaction is defined, for example, does the API gateway need to make a new route when adding a service?